src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -273,6 +288,18 @@ void EnzymesSelectorWidget::setEnzymesList(const QList<SEnzymeData>& enzymes) {
         EnzymeGroupTreeItem* gi = dynamic_cast<EnzymeGroupTreeItem*>(item);
         if (ei != nullptr) {
             teSelectedEnzymeInfo->setHtml(ei->getEnzymeInfo());
+            static constexpr int MAXIMU_CALCULATE_NUMBER_LENGTH = 200'000;
+            if (!ei->hasNumberCalculationTask &&
+                !advSequenceContext.isNull() &&
+                advSequenceContext->getSequenceLength() < MAXIMU_CALCULATE_NUMBER_LENGTH) {
+                auto seqObj = advSequenceContext->getSequenceObject();
+                const auto& er = seqObj->getEntityRef();
+                U2Region reg(0, seqObj->getSequenceLength());
+                auto t = new FindSingleEnzymeTask(er, reg, ei->enzyme, nullptr, seqObj->isCircular(), EnzymeTreeItem::MAXIMUM_ENZYMES_NUMBER, false);
+                AppContext::getTaskScheduler()->registerTopLevelTask(t);
+                connect(t, &FindSingleEnzymeTask::si_stateChanged, this, &EnzymesSelectorWidget::sl_findSingleEnzymeTaskStateChanged);
+                ei->hasNumberCalculationTask = true;
```

Comment:
If I cancel the task via Task manager it will never recalculate the result again with the current implementation.

Reply:
You cannot cancel task via task manager because you have a modal "Restriction enzymes" dialog opened.

Reply:
There are many reasons why cancellation may happen and the dialog should not guess any of them: it should work in its own scope correctly.

Reply:
Same